### PR TITLE
[Bug][issues-2187] if splitTable method execute fast than getNext, job will be blocked

### PR DESCRIPTION
**Pull request by y0908105023**

### CODE REVIEW

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java [hunk 1]
```diff
@@ -291,7 +291,7 @@ public Optional<MySqlSplit> getNext() {
             } else if (!remainingTables.isEmpty()) {
                 try {
                     // wait for the asynchronous split to complete
-                    lock.wait();
+                    lock.wait(100);
```
Comment (ruanhang1993) [inline flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java:294]: Please use a longer timeout(e.g. 4s) and use a constant to maintain this fixed value.

**Review by ruanhang1993 — COMMENTED** [review]
Thanks for the work, @y0908105023 .

**Comment by ruanhang1993** [comment]
Please rebase the master branch to make the CI pass. Thanks.

**Comment by y0908105023** [comment]
Please rebase the master branch to make the CI pass. Thanks.

**Comment by y0908105023** [comment]
Please rebase the master branch to make the CI pass. Thanks.

**Comment by yuxiqian** [comment]
Hi @y0908105023, could you please rebase this PR with latest `master` branch before it could be merged? Thanks!

**Pull request closed without merging**